test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -265,7 +274,13 @@ func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager, options con
 				ToRequests: handler.ToRequestsFunc(r.DockerClusterToDockerMachines),
 			},
 		).
-		WithOptions(options).
+		Watches(
+			&source.Kind{Type: &clusterv1.Cluster{}},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: clusterToDockerMachines,
+			},
+		).
+		WithEventFilter(predicates.ClusterUnpausedAndInfrastructureReady(r.Log)).
```

Comment:
Couldn't we pass the predicate to the above Cluster Watches call? Last argument takes a list of predicates for that watch, instead of adding it as a global event filter.

Reply:
We can use `builder.WithPredicates`

Reply:
It doesn't look like either of those features has made it into a released version of controller-runtime yet, would be more than happy to go that route, though.

Reply:
It would also simplify things by not having to special case non-cluster resources in the predicate handling.